src/Api/Billing/Public/Controllers/OrganizationSubscriptionsController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Net;
+using Bit.Api.Billing.Public.Models;
+using Bit.Api.Models.Public.Response;
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions.Interface;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Billing.Public.Controllers;
+
+[Route("public/organizationsubscriptions")]
```

Comment:
This might just be personal preference, but I think we could simplify this endpoint by making it a bit more RESTful. 

Presumably, our primary resource here is an organization's subscription, and seemingly we just want to update it rather than create one from scratch. As such, I think it would almost be easiest to provide just 1 endpoint:
`PUT /public/organizations/{organizationId}/subscription`

And keep it in an `OrganizationsController` in the same folder. Doing so would let us expand this public API to perform other organization-related billing changes that might not have to do with subscriptions exactly.

This endpoint could have a request body like the following so that the client wouldn't need to invoke two different endpoints to update the same resource:
``` json
{
    "passwordManager": {
        "seats": 1,
        "maxAutoScaleSeats": 10
    },
    "secretsManager": {
        "seats": 1,
        "maxAutoScaleSeats": 10,
        "serviceAccounts": 1,
        "maxAutoScaleServiceAccounts": 2
    },
    "storage": 1
}
```

Reply:
Great suggestion, also this is why you should absolutely get feedback from our integration team as to what API design they prefer. They deal with customers on this stuff every day. And public APIs need to be stable, which means it'll be hard to change whatever the initial implementation here is. (e.g. Jordan C)

Reply:
Also just a cautionary note that some of these properties are nullable (e.g. max autoscale), the API will need to be clear about what happens if a value is not provided. (Does it clear the existing saved value or not update it?)

Reply:
Thanks for the feedback, I have reach out to @notnamed  for prefered API design.

Reply:
Apologies for the delay @cyprain-okeke !

Agreed with the above. Because of the design of the existing Public API endpoint, I would as a customer look under the existing `/public/organization` structure (we currently have `/public/organization/import` which is used in bulk inviting members). However, when it comes to combining the Password Manager and Secrets Manager endpoints, I will defer to someone who's brain works better thinking RESTfully than mine. If I don't have SM enabled in my subscription, is my `secretsManager` JSON object empty, or not present? If I'm adding SM to my PM sub via the API, is that a PUT to this endpoint or a POST?

On the topic of `maxAutoScale`, I would expect a null value to leave my value unchanged, and a -1 to disable the seat limit entirely, so that I don't have to perform a separate call to look up what my existing max is set to. In a PUT request, I think it makes sense to leave the value unchanged - I'm not PUTing a value to that object so keep what is already there.